OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -301,52 +302,46 @@ public void ResolveOrder(Actor self, Order order)
 
 				self.CancelActivity();
 
-				var mobile = self.Trait<Mobile>();
+				CPos? loc;
 				if (order.TargetLocation != CPos.Zero)
 				{
-					var loc = order.TargetLocation;
+					loc = order.TargetLocation;
 					var territory = self.World.WorldActor.TraitOrDefault<ResourceClaimLayer>();
 
 					if (territory != null)
 					{
 						// Find the nearest claimable cell to the order location (useful for group-select harvest):
-						loc = mobile.NearestCell(loc, p => mobile.CanEnterCell(p) && territory.ClaimResource(self, p), 1, 6);
+						loc = mobile.NearestCell(loc.Value, p => mobile.CanEnterCell(p) && territory.ClaimResource(self, p), 1, 6);
 					}
 					else
 					{
 						// Find the nearest cell to the order location (useful for group-select harvest):
 						var taken = new HashSet<CPos>();
-						loc = mobile.NearestCell(loc, p => mobile.CanEnterCell(p) && taken.Add(p), 1, 6);
+						loc = mobile.NearestCell(loc.Value, p => mobile.CanEnterCell(p) && taken.Add(p), 1, 6);
 					}
-
-					self.QueueActivity(mobile.MoveTo(loc, 0));
-					self.SetTargetLine(Target.FromCell(self.World, loc), Color.Red);
-
-					var notify = self.TraitsImplementing<INotifyHarvesterAction>();
-					var next = new FindResources();
-					foreach (var n in notify)
-						n.MovingToResources(self, loc, next);
-
-					LastOrderLocation = loc;
 				}
 				else
 				{
 					// A bot order gives us a CPos.Zero TargetLocation, so find some good resources for him:
-					var loc = FindNextResourceForBot(self);
+					loc = FindNextResourceForBot(self);
 
 					// No more resources? Oh well.
 					if (!loc.HasValue)
 						return;
+				}
 
-					self.QueueActivity(mobile.MoveTo(loc.Value, 0));
-					self.SetTargetLine(Target.FromCell(self.World, loc.Value), Color.Red);
+				self.QueueActivity(new FindResources());
+				self.SetTargetLine(Target.FromCell(self.World, loc.Value), Color.Red);
 
-					LastOrderLocation = loc;
-				}
+				var notify = self.TraitsImplementing<INotifyHarvesterAction>();
+				var next = new FindResources();
```

Comment:
Please move this up a bit and use `self.QueueActivity(next)`, like you do in `UnblockRefinery`.